apps/aecontract/src/aect_state_tree.erl

```diff
@@ -51,24 +51,85 @@ insert_contract(Contract, Tree = #contract_tree{ contracts = CtTree }) ->
     Id         = aect_contracts:id(Contract),
     Serialized = aect_contracts:serialize(Contract),
     CtTree1    = aeu_mtrees:insert(Id, Serialized, CtTree),
-    Tree#contract_tree{ contracts = CtTree1 }.
+    CtTree2    = insert_store(Contract, CtTree1),
+    Tree#contract_tree{ contracts = CtTree2 }.
+
+insert_store(Contract, CtTree) ->
+    Id = aect_contracts:store_id(Contract),
+    Store = aect_contracts:state(Contract),
+    insert_store_nodes(Id, Store, CtTree).
+
+insert_store_nodes(Prefix, Store, CtTree) ->
+    Insert = fun (Key, Value, Tree) ->
+                     Id = <<Prefix/binary, Key/binary>>,
+                     aeu_mtrees:insert(Id, Value, Tree)
+             end,
+     maps:fold(Insert, CtTree, Store).
+
 
 %% @doc Update an existing contract.
 -spec enter_contract(aect_contracts:contract(), tree()) -> tree().
 enter_contract(Contract, Tree = #contract_tree{ contracts = CtTree }) ->
     Id         = aect_contracts:id(Contract),
     Serialized = aect_contracts:serialize(Contract),
     CtTree1    = aeu_mtrees:enter(Id, Serialized, CtTree),
-    Tree#contract_tree{ contracts = CtTree1 }.
+    OldContract = get_contract(Id, Tree),
+    OldStore = aect_contracts:state(OldContract),
+    CtTree2    = enter_store(Contract, OldStore, CtTree1),
```

Comment:
`OldStore`? Shouldn't this be the new store? If not how is the store ever updated?

Reply:
Contract contains new store.
OldStore is used to see if there where keys existing in the store before that has been deleted so that they can be deleted from the tree.

Reply:
Nevermind..